Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -750,10 +756,7 @@ public AndConstraint<TAssertions> Contain(IEnumerable<T> expected, string becaus
         Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot verify containment against a <null> collection");
 
         ICollection<T> expectedObjects = expected.ConvertOrCastToCollection();
-        if (!expectedObjects.Any())
-        {
-            throw new ArgumentException("Cannot verify containment against an empty collection", nameof(expected));
-        }
+        Guard.ThrowIfArgumentIsEmpty(expectedObjects, nameof(expected), "Cannot verify containment against an empty collection");
```

Comment:
I know this is unrelated to the PR, but I just wanted to mention that this validation/exception is very unexpected to see: if feels ilogical.

This would be equivalent to [`string.Contains(string)` throwing if you pass an empty string as an argument. Instead, as expected, it just returns `true`](https://learn.microsoft.com/en-us/dotnet/api/system.string.contains?view=net-7.0#system-string-contains(system-string)), which IMHO is what this method here should do as well: "an empty collection is always contained in any other non-null collection" using the same logic.

Reply:
I can't find the discussions right now, but we decided to not always supports [vacuous truths](https://en.wikipedia.org/wiki/Vacuous_truth).